storage/rocksdb/ha_rocksdb.cc

```diff
@@ -4541,10 +4541,10 @@ static ulonglong rdb_get_int_col_max_value(const Field *field) {
 void ha_rocksdb::load_auto_incr_value() {
   ulonglong auto_incr = 0;
   bool validate_last = false, use_datadic = true;
-#ifndef NDEBUG
+#if !defined(DBUG_OFF)
```

Comment:
I recall pointing out NDEBUG to upstream and them replying that it's fine.

Reply:
I do seem to recall that conversation as well.  This does now cause issues in 8.0 and I do believe it is wrong to use NDEBUG when attempting any kind of conditional compilation that interacts with my_dbug.h.  All functionality of my_dbg.h is based on !DBUG_OFF, therefore, any use of my_dbug.h and external conditional compilation in support of its use should use the same base indicator (DBUG_OFF) and not rely on an assumption that NDEBUG == DBUG_OFF because they are not the same.